pkg/api/api.go

```diff
@@ -225,6 +225,19 @@ type IssueCodeResponse struct {
 	ErrorCode string `json:"errorCode,omitempty"`
 }
 
+// BatchIssueCodeRequest defines the request for issuing many codes at once.
+type BatchIssueCodeRequest struct {
```

Comment:
do we intend for this to be a public API or for use by the UI only.
what are the failure semantics of some work / some don't?

Reply:
Ultimately public, and used by the UI.

The BatchIssueCodeResponse has error fields and a list of individual responses (which also have error codes). The response is error with details if the batch request failed overall. If we get to code processing each individual code has its own errorDetails struct in there. We give status-per-code.

Actually now that you bring it up I'm reading Google's official guidelines: https://google.aip.dev/233 which do not allow partial success.... I'm going to ignore that because I don't want to see SMSes sent twice and I don't think we'll be able to make this easily atomic.

Reply:
I think it's fine to stop at the first failure - but the client has to know where the failure occurred

we can add error level return conditions specific to this API -i.e. cursor, list of already processed UUID, something like that

since the intent is public API - we need strict limits on the batch sizes. I think even starting very low, 5? 10? would give some performance benefit, but make weird failure scenarios less likely

Reply:
right batch size limiting was one of the TODOs I had in there. Put that in and set it to 10 for now. I'm curious what latency will look like with twilio calls and all